### ISSUE TITLE
scylla not working after install.sh

### ISSUE THREAD

**Issue by glommer** [scylladb/scylla#4880]
After 698b72b5018868df6a839d08fd24c642db97ffcd, scylla can't be used from .tar.gz installation.

We have installed it using install.sh to a specific $root, but the trampoline shell script is mentioning absolute paths on /opt/ that can't be resolved.

Just changing these to relative paths is not working either:

```
$ ./usr/bin/scylla
./usr/bin/scylla: line 4: /home/glauber/foo/opt/scylladb/libexec/scylla: No such file or directory
```

The file mentioned in the error message does exist, so we probably made a mistake when rewriting the binary

I am debugging this with @bancron  (user who reported this) on slack

**Comment by avikivity** [comment]
What was the command line given to install.sh?

**Comment by bancron** [comment]
`./install.sh --root /mnt/scylla --target ubuntu --pkg server --disttype debian --python3 /usr/bin/python3`

**Comment by avikivity** [comment]
Don't use --root, that's for chroot style installs. Use --prefix instead (or nothing at all)

**Comment by bancron** [comment]
```
$ ./install.sh --prefix /mnt/scylla --target ubuntu --pkg server --disttype debian --python3 /usr/bin/python3
install: cannot change permissions of ‘//etc/scylla.d’: No such file or directory

$ ./install.sh --target ubuntu --pkg server --disttype debian --python3 /usr/bin/python3
install: cannot change permissions of ‘//etc/scylla.d’: No such file or directory
```

**Comment by avikivity** [comment]
Ugh, /etc is hardcoded and we don't have an override for it.

**Comment by bancron** [comment]
I want to help you make progress on the bugs that will be probably hiding under this one.

I pretended this bug was fixed by symlinking:
```
$ sudo ln -s /home/ubuntu/fake-scylla/etc/scylla.d /etc/scylla.d
$ sudo chown -h ubuntu:ubuntu /etc/scylla.d
```

Great. The next issue is 
```
install: cannot create regular file '//etc/default/scylla-server': Permission denied
```

Hopefully this can be fixed to put the file in the `--prefix` instead of `/etc/default`.

Pretending this is fixed for now by re-running `install.sh` with sudo (which I hope is not supposed to be necessary). Seemed to mostly work; the output mentioned some more absolute paths, but at least they are inside our `--prefix` this time:
```
$ sudo ./install.sh --prefix /mnt/scylla --target ubuntu --pkg server --disttype debian --python3 /usr/bin/python3
warning: working around a Linux kernel bug by creating a hole of 12288 bytes in ‘///mnt/scylla/libexec/gawk’
warning: working around a Linux kernel bug by creating a hole of 319488 bytes in ‘///mnt/scylla/libexec/gzip’
warning: working around a Linux kernel bug by creating a hole of 4096 bytes in ‘///mnt/scylla/libexec/hwloc-calc’
warning: working around a Linux kernel bug by creating a hole of 4096 bytes in ‘///mnt/scylla/libexec/hwloc-distrib’
warning: working around a Linux kernel bug by creating a hole of 4096 bytes in ‘///mnt/scylla/libexec/lscpu’
```

```
$ ls -lah `which scylla`
lrwxrwxrwx 1 root root 27 Aug 23 23:15 /usr/bin/scylla -> ../../mnt/scylla/bin/scylla
$ scylla --version
666.development-0.20190820.698b72b50
```

Let's do it.
```
SCYLLA_CONF=/home/ubuntu/scylla/conf /usr/bin/scylla --log-to-syslog 0 --log-to-stdout 1 --default-log-level info \
                 --network-stack posix --num-io-queues 1 \
                 --options-file /home/ubuntu/fake-scylla/scylla.yaml
Scylla version 666.development-0.20190820.698b72b50 starting ...
command used: "/usr/bin/scylla --log-to-syslog 0 --log-to-stdout 1 --default-log-level info --network-stack posix --num-io-queues 1 --options-file /home/ubuntu/fake-scylla/scylla.yaml"
parsed command line options: [log-to-syslog: 0, log-to-stdout: 1, default-log-level: info, network-stack: posix, num-io-queues: 1, options-file: /home/ubuntu/fake-scylla/scylla.yaml]
WARN  2019-08-23 23:45:32,859 [shard 0] seastar - Unable to set SCHED_FIFO scheduling policy for timer thread; latency impact possible. Try adding CAP_SYS_NICE
INFO  2019-08-23 23:45:32,902 [shard 0] init - installing SIGHUP handler
INFO  2019-08-23 23:45:32,919 [shard 0] init - Scylla version 666.development-0.20190820.698b72b50 starting.
INFO  2019-08-23 23:45:32,920 [shard 0] init - starting prometheus API server
INFO  2019-08-23 23:45:32,921 [shard 0] init - creating tracing
INFO  2019-08-23 23:45:32,921 [shard 0] init - creating snitch
INFO  2019-08-23 23:45:32,923 [shard 0] snitch_logger - Ec2Snitch using region: us-east, zone: 1d.
INFO  2019-08-23 23:45:32,923 [shard 0] init - determining DNS name
INFO  2019-08-23 23:45:32,923 [shard 0] init - starting API server
INFO  2019-08-23 23:45:32,924 [shard 0] init - Scylla API server listening on 127.0.0.1:10000 ...
INFO  2019-08-23 23:45:32,924 [shard 0] init - initializing storage service
INFO  2019-08-23 23:45:32,924 [shard 0] init - starting per-shard database core
ERROR 2019-08-23 23:45:32,927 [shard 0] init - I/O Scheduler is not properly configured! This is a non-supported setup, and performance is expected to be unpredictably bad.
 Reason found: none of --max-io-requests, --io-properties and --io-properties-file are set.
To properly configure the I/O Scheduler, run the scylla_io_setup utility shipped with Scylla.
To ignore this, see the developer-mode configuration option.
INFO  2019-08-23 23:45:32,927 [shard 0] init - Shutdown database started
INFO  2019-08-23 23:45:32,927 [shard 0] init - Shutdown database finished
INFO  2019-08-23 23:45:32,927 [shard 0] init - stopping prometheus API server
INFO  2019-08-23 23:45:32,928 [shard 0] init - Startup failed: std::runtime_error (Bad I/O Scheduler configuration)
```

Alright, maybe we do need to run scylla_setup in order to generate `io_properties.yaml`...
```
$ ls -lah `which scylla_setup`
lrwxrwxrwx 1 root root 37 Aug 23 23:15 /usr/sbin/scylla_setup -> ../../mnt/scylla/scripts/scylla_setup
$ /mnt/scylla/scripts/scylla_setup --no-enable-service --no-ec2-check --no-ntp-setup --no-raid-setup --no-fstrim-setup --nic ens5
Traceback (most recent call last):
  File "/mnt/scylla/scripts/libexec/scylla_setup", line 30, in <module>
    from scylla_util import *
  File "/mnt/scylla/scripts/scylla_util.py", line 32, in <module>
    import psutil
ModuleNotFoundError: No module named 'psutil'
```

Same output with sudo. Okay, nevermind, I'll copy it from a working installation.
```
SCYLLA_CONF=/home/ubuntu/scylla/conf /usr/bin/scylla --log-to-syslog 0 --log-to-stdout 1 --default-log-level info \
                --network-stack posix --num-io-queues 1 \
                --options-file /home/ubuntu/fake-scylla/scylla.yaml \
                --io-properties-file /home/ubuntu/fake-scylla/etc/scylla.d/io_properties.yaml \
                --hints-directory /mnt/scylla/hints --view-hints-directory /mnt/scylla/view_hints
# ... tons of successful-looking log output
```

Looks like it worked! However, I don't think either nodetool or cqlsh were installed as part of this build. Are they supposed to have been?

I hope this helps!

**Comment by glommer** [comment]
Thanks a lot Naomi

psutil is a python-(3) module, so you would have to install that too. Scylla will ship a python3 interpreter that always has all dependencies needed to make this process smoother. You can generate the said interpreter by doing `./reloc/python3/build_reloc.sh` -> same as scylla - and then using that as the python3 interpreter path. This should be optional and if you prefer just install psutil.

THAT SAID: I am pretty sure the scylla_setup will still touch a bunch of absolute paths. I had realized this a couple of weeks ago and started doing work to make it relative before my life descended into a spiral of high latency chaos.

So for now copying the config files from a working installation will help.

@avikivity -> I will leave the /etc change to you

**Comment by slivne** [comment]
this is for offlineinstaller

**Comment by glommer** [comment]
Yes it is.

**Comment by slivne** [comment]
offlineinstaller is not complete for 3.2 - pushing this out

**Comment by slivne** [comment]
Offline installer was released as part of 4.3 - it is working an tested - closing this issue
